client/state/data-layer/wpcom/nps-survey/index.js

```diff
@@ -0,0 +1,73 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import {
+	NPS_SURVEY_CHECK_ELIGIBILITY,
+} from 'state/action-types';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequestEx } from 'state/data-layer/wpcom-http/utils';
+import { setNpsSurveyEligibility } from 'state/nps-survey/actions';
+
+export const fetchSurveyEligibility = action => ( dispatch, getState ) => {
+
+	dispatch(
+		http(
+			{
+				method: 'GET',
+				path: `/nps/`,
+				apiVersion: '1.2',
+				} ),
+			}
+		)
+	);
+};
+
+// @todo this needs to get "data" returned somehow from the API call
+export const handleSuccess = ( action, {} ) => {
```

Comment:
The second argument here is `data`. eg `export const handleSuccess = ( action, data ) => {`

For our convenience we can pull out `display_survey` and rename this from snake_case using destructuring assignment. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

or 
```jsx
export const handleSuccess = ( action, { displaySurvey: display_survey } ) => {
    debug( '...Eligibility returned from endpoint.', data );
	dispatch( setNpsSurveyEligibility( displaySurvey ) );
}
```

To make things even clearer, let's rename the handler to clearly state what action this will take. For example `handleFailure` might be better named as `disableNpsSurvey`

or `handleSuccess` might be:

```jsx
export const updateNpsSurveyEligibility = ( action, { displaySurvey: display_survey } ) => {
    debug( '...Eligibility returned from endpoint.', data );
	dispatch( setNpsSurveyEligibility( displaySurvey ) );
}
```

Reply:
even better can can extract the transformation step into `fromApi` and then it will automatically fail if there's no `display_survey` property in the response

```js
const fromApi = ( { display_survey } ) => display_survey;

const success = ( action, displaySurvey ) => setNpsSurveyEligibility( displaySurvey );
```

Reply:
I was just about to follow up with that @dmsnell ;)

One thing you're missing in the above example is passing it through to dispatchRequestEx.